### PR TITLE
Reorganize development and testing rules

### PR DESCRIPTION
**Pull request by fregante**
I'm moving some config from the extension to this shared config

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -25,4 +27,8 @@ module.exports.rules = {
 
 	// It's a Webpack optimization, not necessary here
 	"import/dynamic-import-chunkname": "off",
+
+	// Disable test rules we don't like
+	"jest/no-export": "off",
+	"testing-library/render-result-naming-convention": "off",
```
Comment (grahamlangford) [inline server.js]: Why would we disable this rule? We've already fixed it in both App and Extension to be compliant iirc.

Comment (fregante) [reply server.js]: I don't know, I just copied it from the extension. I can drop it from both places if it passes

Comment (fregante) [reply server.js]: There are 51 errors in the extension, a PR would need to be sent there to fix them before removing this exception

Comment (grahamlangford) [reply server.js]: I see the problem, we disabled the rule for files that match these patterns: `["**/!(*.test)*.ts?(x)", "**/*.ts"]`, which are some of our test helpers and include render functions from libraries other than testing-library where using `rendered` or `wrapper` wouldn't be an anti-pattern. 

We want to keep the rule on "error" for actual `*.test.ts?(x)` files.

File: index.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const config = {
 		},
 	},
 	ignorePatterns: [".idea", "dist", "**/__mocks__/**"],
-	plugins: ["filenames", "jsx-a11y", "jest", "testing-library"],
+	plugins: ["filenames", "jsx-a11y"],
```
Comment (fregante) [inline index.js:31]: > it's generally not suitable to include them in your top-level configuration

https://github.com/jest-community/eslint-plugin-jest?tab=readme-ov-file#running-rules-only-on-test-related-files

https://github.com/testing-library/eslint-plugin-testing-library?tab=readme-ov-file#run-the-plugin-only-against-test-files

File: index.js [hunk 3]
```diff
@@ -264,6 +272,13 @@ const config = {
 		],
 	},
 	overrides: [
+		{
+			// JS files shouldn't have TypeScript rules, but it's bothersome to separate them properly
+			files: ["**/*.js"],
+			rules: {
+				"@typescript-eslint/no-unsafe-argument": "off",
+			},
+		},
```
Comment (fregante) [inline index.js:281]: Moved from the extension

File: server.js [hunk 4]
```diff
@@ -8,21 +8,14 @@ module.exports.rules = {
...
 	// The webpack/testing files only run on the CI server, so don't need to watch for these:
 	"security/detect-object-injection": "off",
 	"security/detect-child-process": "off",
 	"security/detect-unsafe-regex": "off",
 
-	"jest/no-conditional-expect": "off",
-	"unicorn/consistent-function-scoping": "off",
-
-	// Common and required usage in tests
-	"unicorn/no-useless-undefined": "off",
```
Comment (fregante) [inline server.js:24]: I moved the test-specific rules to tests.js

File: server.js (hunk 1 above)
Comment (fregante) [reply server.js]: Indeed this was a double negation: on _not_ tests, _disable_ 🤔 

Instead, I moved the test rules to run just on test files, as also described in the plugin's readme.

**Review by fregante — COMMENTED** [review]
I opened two PRs to validate this config.

- https://github.com/pixiebrix/pixiebrix-extension/pull/6682
- https://github.com/pixiebrix/pixiebrix-app/pull/4173

**Review by johnnymetz — APPROVED** [review]

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**